File: backend/app/schemas/operator_session.py
"""작업자 세션 API schema."""

from __future__ import annotations

import uuid

from pydantic import BaseModel, Field

from app.schemas.common import UtcDatetime
from app.schemas.employee import EmployeeResponse


class OperatorSessionLoginRequest(BaseModel):
    employee_id: uuid.UUID
    pin: str = Field(..., min_length=4, max_length=4, pattern=r"^[0-9]{4}$")


class OperatorPinChangeCompleteRequest(BaseModel):
    employee_id: uuid.UUID
    new_pin: str = Field(..., min_length=4, max_length=4, pattern=r"^[0-9]{4}$")


class OperatorSessionResponse(BaseModel):
    employee: EmployeeResponse
    expires_at: UtcDatetime
    boot_id: str

File: frontend/lib/api/types/operator-session.ts
import type { Employee } from "./employees";

export interface OperatorSessionResponse {
  employee: Employee;
  server_time: string;
  expires_at: string;
  boot_id: string;
  /** 브라우저가 응답 본문을 받은 시각. 네트워크 뒤 채택되어도 서버 TTL을 늘리지 않는다. */
  client_received_at_ms?: number;
}
